src/main/java/org/apache/lucene/search/vectorhighlight/MultiValueBaseFragmentsBuilder.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2023 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.search.vectorhighlight;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.search.vectorhighlight.FieldFragList.WeightedFragInfo;
+import org.apache.lucene.search.vectorhighlight.FieldFragList.WeightedFragInfo.SubInfo;
+import org.apache.lucene.search.vectorhighlight.FieldPhraseList.WeightedPhraseInfo.Toffs;
+
+public abstract class MultiValueBaseFragmentsBuilder extends BaseFragmentsBuilder {
```

Comment:
Instead of creating a new class `MultiValueBaseFragmentsBuilder` should we override `MultiValueBaseFragmentsBuilder` itself? i.e. Create class with the name `BaseFragmentsBuilder` with the fix. Making similar changes is the reason why we have some classes in the org.apache.lucene package in nrtsearch.